packages/example-codehub/test/acceptance/smoke.test.ts

```diff
@@ -5,10 +5,14 @@
 
 import {CodeHubApplication} from '../../src/codehub-application';
 import {validateApiSpec} from '@loopback/testlab';
+import {RestServer, RestComponent} from '@loopback/rest';
 
-describe('Application', () => {
+describe('Server', () => {
   it('has a valid OpenAPI spec', async () => {
-    const app = new CodeHubApplication();
-    await validateApiSpec(app.getApiSpec());
+    const app = new CodeHubApplication({
+      components: [RestComponent],
```

Comment:
`CodeHubApplication`'s constructor is already adding `RestServer`, it should add `RestComponent` too.

Reply:
There's actually an interesting side-effect from the work I've done. I was playing around with the chit-chat app using this branch as a base and I noticed that I no longer needed to override the Application class with my own custom implementation for typical usage.

Would we want to keep these tests as they currently are, or perhaps alter them to de-emphasize creating your own Application subclass?

Reply:
IMO, we should nudge people towards creating their own Application subclass.

The code accessing the application (typically tests) should not need to know what components etc. the application depends on, that's the business of the custom application class.

Reply:
Now it's possible to configure components etc. even without a custom Application subclass, I don't mind that.

What's important to me is that this app configuration must happens in the production code (e.g. `src/app.js` or `src/main.js`), not in the test files!